usefulUtilities/statusIndicator/webApp/app/statusIndicator_webApp.js

```diff
@@ -192,43 +211,35 @@ function getAllEmployees(response) {
             return response.end(JSON.stringify(responseObject));
         }
 
-        var teamName = "";
-        var responseHTML = "";
-
-        if (results.length === 0) {
-            responseHTML = `<h2>There are no employees!</h2>`;
-        }
+        var responseObject = {
+            "status": "success",
+            "teams": []
+        };
 
+        var currentTeamObject = {};
+        var currentTeamName = false;
         for (var i = 0; i < results.length; i++) {
-            if (teamName !== results[i].teamName) {
-                // different team name
+            if (currentTeamName !== results[i].teamName) {
+                currentTeamName = results[i].teamName;
+
                 if (i > 0) {
-                    // if not the first entry, close the last team
-                    responseHTML += `</table></div>`;
+                    responseObject.teams.push(currentTeamObject);
                 }
-                teamName = results[i].teamName;
-                responseHTML += `<div class="team"><h2>${results[i].teamName}</h2><table>`;
-            }
-
-            var location = results[i].location || "unknown";
 
-            if (results[i].status === "busy") {
-                location = "Hidden";
+                currentTeamObject = {
+                    "name": currentTeamName,
+                    "members": []
+                };
             }
 
-            responseHTML += `
-<tr>
-    <td width="40%">${results[i].displayName}</td>
-    <td width="40%">${results[i].status}</td>
-    <td width="20%">${location}</td>
-</tr>
-            `;
+            currentTeamObject.members.push(formatMemberData(results[i]));
         }
-        responseHTML += `</table></div>`;
+        // Push the last one.
+        responseObject.teams.push(currentTeamObject);
```

Comment:
This must be the false one then?

Reply:
i'm not quite sure what you mean here. are you saying "this must be the object in which the team name is false" ? if so, no, that's not true, because the team name is never false.

Reply:
Gotcha, was just not sure why that was pushed separately outside of the for loop?

Reply:
<discussed offline>